lib/still/preprocessor.ex

```diff
@@ -10,18 +13,60 @@ defmodule Still.Preprocessor do
     OutputPath,
     OutputPath,
     Slime,
-    URLFingerprinting
+    URLFingerprinting,
+    Save,
+    AddLayout,
+    AddContent
   }
 
   @default_preprocessors %{
-    ".slim" => [Frontmatter, Slime, OutputPath],
-    ".slime" => [Frontmatter, Slime, OutputPath],
-    ".eex" => [Frontmatter, EEx, OutputPath],
-    ".css" => [EEx, CSSMinify, OutputPath, URLFingerprinting],
-    ".js" => [EEx, JS, OutputPath, URLFingerprinting],
-    ".md" => [Frontmatter, EEx, Markdown, OutputPath]
+    ".slim" => [AddContent, Frontmatter, Slime, OutputPath, AddLayout, Save],
+    ".slime" => [AddContent, Frontmatter, Slime, OutputPath, AddLayout, Save],
+    ".eex" => [AddContent, Frontmatter, EEx, OutputPath, AddLayout, Save],
+    ".css" => [AddContent, EEx, CSSMinify, OutputPath, URLFingerprinting, AddLayout, Save],
+    ".js" => [AddContent, EEx, JS, OutputPath, URLFingerprinting, AddLayout, Save],
```

Comment:
Now that everything is a preprocessor, imagine I want to add my own custom preprocessor. In the settings I can't just set

```elixir
config :still, preprocessors: %{ ".js" => MyJSPreprocessor}
```

I need to wrap in `AddContent, AddLayout, OutputPath, URLFingerprinting, AddLayout, Save`, etc, don't I?

Reply:
Yes, for now. I think it's a bit early to introduce an abstraction. Even now that I'm implementing image preprocessing I'm finding some limitations with the existing API. Not with this part, but with the include part. I would rather do it once we have a couple of sites implemented just to be sure the foundations are right.

Reply:
As a developer that will be a huge source of problems and quite frankly a deal breaker. We should enforce these. At the very least, this should do the trick:

```elixir
def get_preprocessor_for(ext) do
  preprocessors = get_preprocessor_for_config(ext) || []

  [AddContent, EEx, preprocessors, OutputPath, URLFingerprinting, AddLayout, Save]
  |> List.flatten()
  |> Enum.uniq()
end
```

Reply:
We discussed this and agreed on pushing back due to further pending changes that will soon hit master